examples/step-81/step-81.cc

```diff
@@ -0,0 +1,893 @@
+                const auto id = face->boundary_id();
+                if (id != 0)
+                  {
+                    fe_face_values.reinit(cell, face);
+                    FEValuesViews::Vector<dim> real_part(fe_face_values, 0);
+                    FEValuesViews::Vector<dim> imag_part(fe_face_values, dim);
+
+                    for (unsigned int q_point = 0; q_point < n_face_q_points;
+                         ++q_point)
+                      {
+                        const auto &position = quadrature_points[q_point];
+                        const auto  radius   = position.norm();
+                        const auto  inner_radius =
+                          perfectly_matched_layer.inner_radius;
+
+                        auto mu_inv  = parameters.mu_inv(position, id);
+                        auto epsilon = parameters.epsilon(position, id);
+
+                        if (radius >= inner_radius)
+                          {
+                            auto A = perfectly_matched_layer.a_matrix(position);
+                            auto B = perfectly_matched_layer.b_matrix(position);
+                            auto d = perfectly_matched_layer.d(position);
+
+                            mu_inv  = mu_inv / d;
+                            epsilon = invert(A) * epsilon * invert(B);
+                          };
+
+                        const auto normal =
+                          fe_face_values.normal_vector(q_point);
+
+                        for (unsigned int i = 0; i < dofs_per_cell; ++i)
+                          {
+                            const auto phi_i =
+                              real_part.value(i, q_point) -
+                              1.0i * imag_part.value(i, q_point);
+                            const auto phi_i_T = tangential_part(phi_i, normal);
+
+                            for (unsigned int j = 0; j < dofs_per_cell; ++j)
+                              {
+                                const auto phi_j =
+                                  real_part.value(j, q_point) +
+                                  1.0i * imag_part.value(j, q_point);
+                                const auto phi_j_T =
+                                  tangential_part(phi_j, normal) *
+                                  fe_face_values.JxW(q_point);
+
+                                const auto prod      = mu_inv * epsilon;
+                                const auto sqrt_prod = prod;
+
+                                const auto temp =
+                                  -1.0i * scalar_product((sqrt_prod * phi_j_T),
+                                                         phi_i_T);
+                                cell_matrix(i, j) += temp.real();
+                              } /* j */
+                          }     /* i */
+                      }         /* q_point */
+                  }
+              }
+            else
+              {
+                Assert(!face->at_boundary(), ExcMessage("oops!"));
```

Comment:
Do we need this statement? I would suggest removing it, because that's precisely the case we do in the `if` clause and I see no way this would add information, apart from the initial code debugging that has long been concluded.